simplex.py

```diff
@@ -195,25 +197,20 @@ def run(self, r_data=None):
             ref_ff = self.ff.__class__()
             ref_ff.method = 'REFLECTION'
             ref_ff.params = copy.deepcopy(best_ff.params)
+            # Need score difference sum for weighted inversion. If zero, should break
+            score_diff_sum=sum([x.score - self.new_ffs[-1].score
+                                 for x in self.new_ffs[:-1]])
+            if score_diff_sum==0.:
+                logger.warning(
+                    'No difference between force field scores, breaking.')
+                break
             for i in xrange(0, len(best_ff.params)):
                 if self.do_weighted_reflection:
-                    try:
```

Comment:
What you added in, the `score_diff_sum` is the same as what I have here.

Yours 1st checks whether that is zero, whereas mine simply breaks if it is zero. However, your method simply breaks without writing the best FF. This is bad and will break simplex from writing out the best current results. An error must be raised, such that `opt.catch_run_errors` can figure out how to handle the situation.